api/models/model_class_possibilities/Model_id_recursion.py

```diff
@@ -0,0 +1,60 @@
+# for more detail, see class Model in :
+# https://github.com/pallets/flask-sqlalchemy/blob/master/flask_sqlalchemy/model.py
+from typing import Any, List, Type
+
+
+class MyModel(object):
+    query_class = None  # flask_alchemy attribute
+    query = None  # flask_alchemy attribute
+
+    DONOTSEND_MODEL = {'_sa_instance_state'}
+    DONOTSEND = []
+    _jsonified = None
+
+    def __repr__(self) -> str:
+        return '<{}>'.format(self.__class__.__name__)
+
+    def to_dict_recursive(self) -> dict:
+        return self._to_dict_recursive(list_objects_id_passed_through=[id(self)])
+
+    def _to_dict_recursive(self, list_objects_id_passed_through: List[int]) -> dict:
+        # functions :
+        # anti_circular_recursion : check if we've already called the object
+        #                               if not do the recursion
+        # type_shunt_recursive : select the actions for each type of attr
+
+        def anti_circular_recursion(obj: Type[MyModel]) -> any:
+            if id(obj) in list_objects_id_passed_through:
+                return str(obj)
+                # others possibilities
+                # return str(obj).join(' ').join(str(obj.id))
+                # return obj.id
+            else:
+                list_objects_id_passed_through.append(id(obj))
+                return obj._to_dict_recursive(list_objects_id_passed_through)
+
+        ##### what about dict which contains object(s) ? Is it possible in SQLAlchemy ?
```

Comment:
this wouldn't be required to be handled! sql tables are dictionaries :)

Reply:
perfect then :]